providers/default.rb

```diff
@@ -496,14 +496,16 @@ def execute_install_script(install_script)
                         ''
                       end
 
-    post_action = if new_resource.post_install_action == 'exec'
+    license_provided = node['chef_client']['chef_license'] || ''
+
+    post_action = if (new_resource.post_install_action == 'exec') && (desired_version.major >= 15)
+                    "#{new_resource.exec_command} --chef-license #{license_provided}"
+                  elsif desired_version.major < '15'
                     new_resource.exec_command
                   else
                     ''
                   end
```

Comment:
```suggestion
    post_action = if new_resource.post_install_action != 'exec'
                             ''
                           elsif desired_version.major >= 15
                             "#{new_resource.exec_command} --chef-license #{license_provided}"
                           else
                             new_resource.exec_command
                           end
```

I think this conditional needs to change to maintain the old behavior. 

The whitespace is probably all wrong because I'm trying to do this in github's comment box.

Reply:
Updated. Thank you.